src/app/screens/Settings.tsx

```diff
@@ -1,36 +1,109 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import browser from "webextension-polyfill";
 import { Html5Qrcode } from "html5-qrcode";
 
 import Container from "../components/Container";
 import Button from "../components/Button";
+import Toggle from "../components/Toggle";
+
+type Props = {
+  title: string;
+  subtitle: string;
+  right: React.ReactNode;
+};
+
+function Setting({ title, subtitle, right }: Props) {
+  return (
+    <div className="py-4 flex justify-between items-center">
+      <div>
+        <span className="text-gray-700 font-medium">{title}</span>
+        <p className="text-gray-400 text-sm">{subtitle}</p>
+      </div>
+      {right}
+    </div>
+  );
+}
 
 function Settings() {
+  const [loading, setLoading] = useState(true);
+  const [settings, setSettings] = useState<{ websiteEnhancements: boolean }>({
+    websiteEnhancements: false,
+  });
   const [cameraPermissionsGranted, setCameraPermissionsGranted] =
     useState(false);
 
+  function getSettings() {
+    browser.storage.sync
```

Comment:
I think this should go through an `action` and the state. what do you think? 

in the end it is a `state.setState({})`

Reply:
Agree, will update :)